xinference/model/rerank/core.py

```diff
@@ -0,0 +1,224 @@
+        return {
+            "model_type": "rerank",
+            "address": self.address,
+            "accelerators": self.devices,
+            "model_name": self._model_spec.model_name,
+            "language": self._model_spec.language,
+            "model_revision": self._model_spec.model_revision,
+        }
+
+
+class RerankModel:
+    def __init__(
+        self,
+        model_uid: str,
+        model_path: str,
+        device: Optional[str] = None,
+        model_config: Optional[Dict] = None,
+    ):
+        self._model_uid = model_uid
+        self._model_path = model_path
+        self._device = device
+        self._model_config = model_config or dict()
+        self._model = None
+
+    def load(self):
+        try:
+            from sentence_transformers.cross_encoder import CrossEncoder
+        except ImportError:
+            error_message = "Failed to import module 'SentenceTransformer'"
+            installation_guide = [
+                "Please make sure 'sentence-transformers' is installed. ",
+                "You can install it by `pip install sentence-transformers`\n",
+            ]
+
+            raise ImportError(f"{error_message}\n\n{''.join(installation_guide)}")
+        self._model = CrossEncoder(
+            self._model_path, device=self._device, **self._model_config
+        )
+
+    def rerank(
+        self,
+        documents: List[str],
+        query: str,
+        top_n: Optional[int],
+        max_chunks_per_doc: Optional[int],
+        return_documents: Optional[bool],
+    ) -> Rerank:
+        assert self._model is not None
+        if max_chunks_per_doc is not None:
+            raise ValueError("rerank hasn't support `max_chunks_per_doc` parameter.")
+        sentence_combinations = [[query, doc] for doc in documents]
+        similarity_scores = self._model.predict(sentence_combinations)
+        sim_scores_argsort = list(reversed(np.argsort(similarity_scores)))
+        if top_n is not None:
+            sim_scores_argsort = sim_scores_argsort[:top_n]
+        if return_documents:
+            docs = [
+                DocumentObj(
+                    index=int(arg),
+                    relevance_score=float(similarity_scores[arg]),
+                    document=Document(text=documents[arg]),
+                )
+                for arg in sim_scores_argsort
+            ]
+        else:
+            docs = [
+                DocumentObj(
+                    index=int(arg),
+                    relevance_score=float(similarity_scores[arg]),
+                    document=None,
+                )
+                for arg in sim_scores_argsort
+            ]
+        return Rerank(id=str(uuid.uuid1()), results=docs)
+
+
+def get_cache_status(
+    model_spec: RerankModelSpec,
+) -> bool:
+    cache_dir = os.path.realpath(
+        os.path.join(XINFERENCE_CACHE_DIR, model_spec.model_name)
+    )
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir, exist_ok=True)
```

Comment:
I noticed that the `get_cache_status` should not create an empty cache dir. I will create a PR to fix the `get_cache_status` in other model types, could you remove this makedirs in this PR?

Reply:
Done.